frontend/src/js/components/PageViewer/Controls.tsx

```diff
@@ -1,40 +1,117 @@
-import React, { FC } from "react";
-import RotateLeft from "react-icons/lib/md/rotate-left";
-import RotateRight from "react-icons/lib/md/rotate-right";
-import styles from "./Controls.module.css";
-import { FindInput } from "./FindInput";
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import RotateLeft from 'react-icons/lib/md/rotate-left';
+import RotateRight from 'react-icons/lib/md/rotate-right';
+import styles from './Controls.module.css';
+import { FindInput } from './FindInput';
 import { HighlightForSearchNavigation } from './model';
+import { removeLastUnmatchedQuote } from '../../util/stringUtils';
+import authFetch from '../../util/auth/authFetch';
+import { HighlightsState } from './PageViewer';
 
 type ControlsProps = {
   // Rotation
   rotateClockwise: () => void;
   rotateAnticlockwise: () => void;
 
-  // Find Search Input
-  findSearch: string;
-  setFind: (v: string) => void;
-
-  performFind: (query: string) => Promise<void>;
-  isPending: boolean;
-
-  jumpToNextFindHit: () => void;
-  jumpToPreviousFindHit: () => void;
-  findHighlights: HighlightForSearchNavigation[];
-  focusedFindHighlightIndex: number | null;
+  uri: string;
+  onHighlightStateChange: (newState: HighlightsState) => void;
+  onQueryChange: (newQuery: string) => void;
 };
 
 export const Controls: FC<ControlsProps> = ({
   rotateClockwise,
   rotateAnticlockwise,
-  findSearch,
-  setFind,
-  jumpToNextFindHit,
-  jumpToPreviousFindHit,
-  performFind,
-  isPending,
-  findHighlights,
-  focusedFindHighlightIndex,
+  uri,
+  onHighlightStateChange,
+  onQueryChange
 }) => {
+  const [focusedFindHighlightIndex, setFocusedFindHighlightIndex] = useState<number | null>(null);
+  const [findHighlights, setFindHighlights] = useState<HighlightForSearchNavigation[]>([]);
+  // TODO: should we use ths?
+  const [, setFindVisible] = useState(false);
+  const [isFindPending, setIsFindPending] = useState<boolean>(false);
+
+  useEffect(() => {
+    onHighlightStateChange({
+      focusedIndex: focusedFindHighlightIndex,
+      highlights: findHighlights
+    });
+  }, [focusedFindHighlightIndex, findHighlights, onHighlightStateChange])
+
+
+  const performFind = useCallback((query: string) => {
+    if (!query) {
+      setFocusedFindHighlightIndex(null);
+      setFindHighlights([]);
+      onQueryChange(query);
+      return;
+    }
+
+    const params = new URLSearchParams();
+    // The backend will respect quotes and do an exact search,
+    // but if quotes are unbalanced elasticsearch will error
+    params.set("q", removeLastUnmatchedQuote(query));
+
+    // In order to use same debounce on communicating query change to parent
+    onQueryChange(query);
+    setIsFindPending(true);
+    return authFetch(`/api/pages2/${uri}/find?${params.toString()}`)
+      .then((res) => res.json())
+      .then((highlights) => {
+        setIsFindPending(false);
+        setFindHighlights(highlights);
+        if (highlights.length) {
```

Comment:
maybe length !== 0 in case it ends up being undefined e.g. if json is not an array